src/Data/Singletons/Sigma.hs

```diff
@@ -60,6 +61,18 @@ projSigma1 (a :&: _) = fromSing a
 projSigma2 :: forall s t r. (forall (fst :: s). t @@ fst -> r) -> Sigma s t -> r
 projSigma2 f ((_ :: Sing (fst :: s)) :&: b) = f @fst b
 
+-- | Witnessing an adjunction between 'Sigma' and 'ConstSym1'.
```

Comment:
While this is probably true, I don't think it's the most accessible documentation we could give this function. My inclination would be to adapt the Haddocks for the `curry` function in the `Prelude`. Something like:

```haskell
-- | Convert a dependent, uncurried function to a curried one.
```

Similarly for the Haddocks for `uncurrySigma`.